core-requirements.txt

```diff
@@ -3,7 +3,6 @@ pandas>=1.2.1
 scipy>=1.3.3
 scikit-learn>=0.24.0
 scikit-optimize>=0.8.1
-pyzmq>=20.0.0,<22.0.0
```

Comment:
Would also suggest removing it from here
- https://github.com/alteryx/evalml/blob/main/evalml/tests/dependency_update_check/minimum_core_requirements.txt#L6